From 0b05c641626d2742b393c9771064482fc80e48bb Mon Sep 17 00:00:00 2001
From: genspark_dev <genspark_dev@genspark.ai>
Date: Mon, 7 Sep 2026 14:24:02 +0000
Subject: [PATCH] fix: bump Gradle to 9.1.0 and AGP to 8.11.1 for Flutter
 3.47.2 (KGP 2.3.20)

- gradle wrapper: 8.14 -> 9.1.0 (Flutter warning threshold)
- settings.gradle.kts: AGP 8.9.1 -> 8.11.1 (Flutter min), Kotlin 2.1.0 -> 2.3.20 (Gradle 9 compatible)
- app/build.gradle.kts: replace deprecated kotlinOptions with kotlin { compilerOptions { jvmTarget } }
- android.newDsl=false already present in both android trees (AGP < 9 safety net)
---
 android/app/build.gradle.kts                          | 11 +++++++----
 android/gradle/wrapper/gradle-wrapper.properties      |  2 +-
 android/settings.gradle.kts                           |  4 ++--
 styledrop/android/app/build.gradle.kts                | 11 +++++++----
 .../android/gradle/wrapper/gradle-wrapper.properties  |  2 +-
 styledrop/android/settings.gradle.kts                 |  4 ++--
 6 files changed, 20 insertions(+), 14 deletions(-)

diff --git a/android/app/build.gradle.kts b/android/app/build.gradle.kts
index 0b034ab..21f922c 100644
--- a/android/app/build.gradle.kts
+++ b/android/app/build.gradle.kts
@@ -1,5 +1,6 @@
 import java.util.Properties
 import java.io.FileInputStream
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 
 plugins {
     id("com.android.application")
@@ -24,10 +25,6 @@ android {
         isCoreLibraryDesugaringEnabled = true
     }
 
-    kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_11.toString()
-    }
-
     defaultConfig {
         applicationId = "com.styledrop.app"
         minSdk = flutter.minSdkVersion
@@ -58,6 +55,12 @@ android {
     }
 }
 
+kotlin {
+    compilerOptions {
+        jvmTarget.set(JvmTarget.JVM_11)
+    }
+}
+
 flutter {
     source = "../.."
 }
diff --git a/android/gradle/wrapper/gradle-wrapper.properties b/android/gradle/wrapper/gradle-wrapper.properties
index e4ef43f..2d428bf 100644
--- a/android/gradle/wrapper/gradle-wrapper.properties
+++ b/android/gradle/wrapper/gradle-wrapper.properties
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.14-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-9.1.0-all.zip
diff --git a/android/settings.gradle.kts b/android/settings.gradle.kts
index fb605bc..e68f63e 100644
--- a/android/settings.gradle.kts
+++ b/android/settings.gradle.kts
@@ -19,8 +19,8 @@ pluginManagement {
 
 plugins {
     id("dev.flutter.flutter-plugin-loader") version "1.0.0"
-    id("com.android.application") version "8.9.1" apply false
-    id("org.jetbrains.kotlin.android") version "2.1.0" apply false
+    id("com.android.application") version "8.11.1" apply false
+    id("org.jetbrains.kotlin.android") version "2.3.20" apply false
 }
 
 include(":app")
diff --git a/styledrop/android/app/build.gradle.kts b/styledrop/android/app/build.gradle.kts
index 0b034ab..21f922c 100644
--- a/styledrop/android/app/build.gradle.kts
+++ b/styledrop/android/app/build.gradle.kts
@@ -1,5 +1,6 @@
 import java.util.Properties
 import java.io.FileInputStream
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 
 plugins {
     id("com.android.application")
@@ -24,10 +25,6 @@ android {
         isCoreLibraryDesugaringEnabled = true
     }
 
-    kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_11.toString()
-    }
-
     defaultConfig {
         applicationId = "com.styledrop.app"
         minSdk = flutter.minSdkVersion
@@ -58,6 +55,12 @@ android {
     }
 }
 
+kotlin {
+    compilerOptions {
+        jvmTarget.set(JvmTarget.JVM_11)
+    }
+}
+
 flutter {
     source = "../.."
 }
diff --git a/styledrop/android/gradle/wrapper/gradle-wrapper.properties b/styledrop/android/gradle/wrapper/gradle-wrapper.properties
index e4ef43f..2d428bf 100644
--- a/styledrop/android/gradle/wrapper/gradle-wrapper.properties
+++ b/styledrop/android/gradle/wrapper/gradle-wrapper.properties
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.14-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-9.1.0-all.zip
diff --git a/styledrop/android/settings.gradle.kts b/styledrop/android/settings.gradle.kts
index fb605bc..e68f63e 100644
--- a/styledrop/android/settings.gradle.kts
+++ b/styledrop/android/settings.gradle.kts
@@ -19,8 +19,8 @@ pluginManagement {
 
 plugins {
     id("dev.flutter.flutter-plugin-loader") version "1.0.0"
-    id("com.android.application") version "8.9.1" apply false
-    id("org.jetbrains.kotlin.android") version "2.1.0" apply false
+    id("com.android.application") version "8.11.1" apply false
+    id("org.jetbrains.kotlin.android") version "2.3.20" apply false
 }
 
 include(":app")
-- 
2.43.0

